### PR TITLE
Prepare changelog for 1.17.0 release

### PR DESCRIPTION
**Pull request by jack-berg**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,55 @@
...
+
+* BREAKING: delete `LogDataBuilder`. A similar implementation of `LogData` called `TestLogData` has
+  been added to `opentelemetry-sdk-logs-testing`.
+* BREAKING: rename `LogProcessor#emit(LogData)` to `LogProcessor#onEmit(ReadWriteLogRecord)`. The
+  argument change from `LogData` to `ReadWriteLogRecord` allows implementations to mutate logs. To
+  obtain `LogData`, call `ReadWriteLogRecord#toLogData()`.
+* Optimize `SdkLogEmitterProvider` to return noop `LogEmitter` when no `LogProcessor`s are
+  registered.
+
+#### Exporter
```
Comment (rapphil) [inline CHANGELOG.md:35]: Is it necessary to include in the changelog these changes as they are noticeable by users:
* https://github.com/open-telemetry/opentelemetry-java/commit/d84a111273801a3b1be13dc7312379a2fc7266f6
* https://github.com/open-telemetry/opentelemetry-java/commit/4bfe17fb76c291f774de30e53cd3698e89e50adc

Comment (jack-berg) [reply CHANGELOG.md:35]: I omitted those because I think a very small percentage of users will notice them.

Comment (jkwatson) [reply CHANGELOG.md:35]: I think I'd err on the side of over-communicating in this case, but I don't feel super strongly about it.

**Review by jkwatson — APPROVED** [review]

**Review by mateuszrzeszutek — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,60 @@
...
+#### SDK Extensions
+
+* Extend View file based configuration with support for specifying explicit bucket histogram bucket
+  boundaries and exponential bucket counts.
+* Extend autoconfigure SPI `AutoConfigurationCustomizerProvider` and `ResourceProvider` with option
+  to specify ordering.
+* Add autoconfigure SPI with `ConfigurableLogExporterProvider`, allowing custom named log exporters
+  to be provided and selected via autoconfigure.
+* Extend autoconfigure SPI with `AutoConfigurationCustomizer#addPropertiesCustomizer`, providing the
+  ability examine current configuration properties and add / overwrite properties.
```
Comment (trask) [inline CHANGELOG.md:57]: ❤️

**Review by trask — APPROVED** [review]

**Pull request merged**